Source/DataSources/KmlDataSource.js

```diff
@@ -255,6 +255,33 @@ define([
         return deferred.promise;
     }
 
+    function insertNamespaces(text) {
+        var namespaceMap = {
+            xsi : 'http://www.w3.org/2001/XMLSchema-instance'
+        };
+        var firstPart, lastPart, reg, declaration;
+
+        for (var key in namespaceMap) {
+            if (namespaceMap.hasOwnProperty(key)) {
+                reg = RegExp('[< ]' + key + ':');
+                declaration = 'xmlns:' + key + '=';
+                if (reg.test(text) && text.indexOf(declaration) === -1) {
+                    if (!firstPart) {
```

Comment:
This should be
```
if (!defined(firstPart)) {
```

Reply:
do you mean this line should actually use 'defined'? or just check if it's defined?

Reply:
`defined` is a function in Cesium. It does some extra checking to make sure a value is actually `undefined` or `null`. This should be used in cases like this.

The magic of javascript's implicit type conversion would cause `!firstpart` to be true in other cases.